code/modules/overmap/ships/computers/comms.dm

```diff
@@ -0,0 +1,247 @@
+/obj/machinery/computer/ship/comms
+	name = "ship communication console"
+	icon_keyboard = "generic_key"
+	icon_screen = "helm"
+	light_color = "#7faaff"
+	core_skill = SKILL_DEVICES
+
+// TODO: add enabling or disabling comms transponder to hide transmission location
+// TODO: use comms console to target a specific entity with a specific maser rather than targetting everyone in range
+/obj/machinery/computer/ship/comms/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+
+	if(!linked)
+		display_reconnect_dialog(user, "Communication")
+		return
+
+	var/data[0]
+
+	var/list/printing_entities = list()
+	var/list/entities = list()
+	var/list/broadcasters = list()
+	var/list/receivers = list()
+
+	for(var/obj/machinery/shipcomms/comms in linked.comms_masers)
+		var/functional = (comms.stat & (BROKEN|NOPOWER))
```

Comment:
Won't this set functional to true if the comms are broken/have no power?
Maybe it's suppose to be `!(comms.stat & (BROKEN|NOPOWER))` but then again below you do have `"desc" = (functional ? "inactive" : "active")` but then you do an `if(functional)` and do a bunch of work so maybe I am lost. I don't think the var name indicates the correct behavior